### ISSUE TITLE
Navigation bar shortcut key bug

### ISSUE THREAD

**Issue by Alfred-Ashin** [gluonhq/scenebuilder#554]
Hello Dr. AlmasB, can you cancel the shortcut keys of the navigation bar. When searching for components, the shortcut keys of the navigation bar are often triggered, which is very annoying; for example, when I want to search for the StackPane component, when I enter the letter a It will trigger the Arrange navigation bar; input I when entering ImageView will trigger the Insert navigation bar; it makes people unable to continue searching; the version I use is 18.0.0.thanks!

**Comment by xialote** [comment]
I also encountered this problem, often triggered the navigation bar shortcut keys, resulting in query, modify control properties input difficulties.

**Comment by 974988176** [comment]
It seems that the 18 version has this problem. The 8 version does not. I have 2 versions installed on my computer. The latest 18 version has this problem. It conflicts with the navigation bar shortcut key

**Comment by Oliver-Loeffler** [comment]
Hi @Alfred-Ashin,
I tried to reproduce the issue on Windows, there the search only works when focus is at the Library text field.
Also on macOS, same behavior. There I can only reproduce your problem, when I accidently press `ALT`, `SHIFT` or `CTRL`  or the mac specific modifier keys.

Can you please describe on which operating system this happened?

Thanks!

**Comment by 974988176** [comment]
@Oliver-Loeffler 
My operating system is windows, appear in this problem, first you press and raise the ALT key and the menu bar in the shortcuts, said you can again according to the first letters to activate the navigation bar, if at this time, again did not activate the menu bar or press the ALT key, which can lead to search in the components, the navigation error of activated. The problem is that the navigation bar is incorrectly listening for events, and the user only wants to activate the navigation bar by pressing the first letter of the navigation bar while holding ALT and not lifting it, rather than pressing ALT and then pressing the navigation bar. I'd like to change the navigation events so that they only take effect when ALT is held down and disappear when ALT is lifted

**Comment by Oliver-Loeffler** [comment]
Mhh, this seems to be related to issue #250 .

**Comment by 974988176** [comment]
@Oliver-Loeffler YES, these are the same issue

**Comment by Oliver-Loeffler** [comment]
Well, Scene Builder here exactly behaves like other Windows applications (like Pint, Notepad, Word etc.).
1x pressing ALT makes the menubar keyboard navigatable. Of course, this collides with search for components. So this can be clumsy.

Following behavior could help.
1. focus at search text field - `ALT` is ignored
2. when pressing `ESC` or `TAB` focus is moved away from text field and ALT is continued to be accepted

What do you think?

**Comment by 974988176** [comment]
@Oliver-Loeffler hello,hold Alt and F to take effect shortcut is hard do that? I think most people don't like the way this is a separate combination of shortcuts

**Comment by Oliver-Loeffler** [comment]
Good idea! But ALT+F collides with the file menu, ALT+F is pretty much standard on Windows for english language.
How about CTRL+F? CTRL+F would jump into the text field, block the ALT and one could search. ESC or TAB would exit the search.

**Comment by Oliver-Loeffler** [comment]
CTRL is also used for other features such as preview window (CTRL+P) so it would kind of blend in.

**Comment by 974988176** [comment]
@Oliver-Loeffler How about removing navigation shortcuts？

**Comment by Oliver-Loeffler** [comment]
I don't think, that this is a good idea. People are using those shortcuts and they are using the keyboard generally. Removing those would limit the accessibility of Scene Builder. This is something I would like to avoid in general.

@AlmasB what do you think? Any thoughts on this, ideas?

**Comment by 974988176** [comment]
@Oliver-Loeffler Can I set Alt to deactivate navigation when it is raised? Now, as long as you press Alt, it stays active

**Comment by Oliver-Loeffler** [comment]
Not sure on that.

**Comment by 974988176** [comment]
@Oliver-Loeffler 
I guess the navigation shortcut is written like this

```java
KeyCombination kc1 = new KeyCodeCombination(KeyCode.F, KeyCombination.ALT_DOWN);
        Mnemonic mnemonic1 = new Mnemonic(bu1,kc1);
        scene.addMnemonic(mnemonic1);
```
However, this will cause the Alt key to be released and the F press alone will trigger again.
Here's my recommended way of writing that doesn't have this problem.
```java
KeyCombination kccb = new KeyCodeCombination(KeyCode.F, KeyCombination.ALT_DOWN);
        scene.getAccelerators().put(kccb, () -> {
            bu1.fire(); //manually triggers the BUT1 button event
        });
```

**Comment by pitzzahh** [comment]
I think I have the same problem, when trying to search for library,  setting properties such as a style class or code event names, it triggers the menu at the top of the scene builder. So I decided to use the Scene builder directly from IntelliJ which is also great but I prefer the desktop scene builder.
